addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueStateColorBulb.java

```diff
@@ -78,17 +78,65 @@ public HueStateColorBulb(HSBType hsb) {
      * Converts this HueState to a HSBType
      */
     public HSBType toHSBType() {
-        int bri = this.bri * 100 / MAX_BRI;
-        int sat = this.sat * 100 / MAX_SAT;
-        int hue = this.hue * 360 / MAX_HUE;
-
-        if (!this.on) {
-            bri = 0;
-        }
-        if (colormode == ColorMode.ct) {
-            sat = 0;
+        if (colormode == ColorMode.xy) {
```

Comment:
This looks like some complex piece of code which is not really completely covered in that single touched test.

Reply:
That is indeed correct, that is because the colors can never be converted completely accurately, we're talking about inputs with many decimal places here. The approach here was "as long as we get a color that could be close, it's fine, as this conversion will never be 100% accurate anyways". The problem is that we can't say "if we input 1 and 0 we expect color code #124522 to come out", but instead we want something that is very close to that color (again we're talking about 2 completely different approaches of color-definition here). I honestly don't know if I should add a testcase to check the output here.

Reply:
I'm not into color conversion but I can image that there are corner cases (the code seems at least to contain a lot).

If others think that its good enough without then I'm okay accepting it. But the combination of complex and untested is generally not something I prefer (in the week before the release even more)

Reply:
Could you maybe pick 2 valid pairs of x,y compute hsb with it and then just compare the computed value to the algorithms output? That way we can catch if anybody messes up with this computation in the future.